### PR TITLE
REF: handling of max_colwidth parameter

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

this would be a precursor to fixes for #7059, #9784 and #6491.

getting the display.max_colwidth value is moved from `_make_fixed_width` to `format_array` to be consistent with the handling of the `column_space`, `float_format` and `precision` display options.

the PR also includes refactoring of the prior fix to `to_html` #24841

`_format_col` is moved from `DataFrameFormatter` to `TableFormatter` to make it available to `HTMLFormatter` and `NotebookFormatter`.

at the moment `_make_fixed_width` is called by `format_array`, but also by `_to_str_columns` and `_get_formatted_index`.

`_to_str_columns` is not an issue because the calls are preceded to calls to `format_array`

IMO `_get_formatted_index` needs to be refactored to call `format_array` directly and probably some additional tests as a precursor to this PR.

thoughts?

cc @TomAugspurger

### CODE REVIEW

**Comment by jreback** [comment]
can you merge master and update

File: pandas/io/formats/format.py [hunk 1]
```diff
@@ -370,6 +370,15 @@ def _get_formatter(self, i):
                 i = self.columns[i]
             return self.formatters.get(i, None)
 
+    def _format_col(self, i):
+        frame = self.tr_frame
```
Comment (jreback) [inline pandas/io/formats/format.py]: can you add a doc-string here (types annotations also appreciated! but just do if you can)

File: pandas/io/formats/format.py [hunk 2]
```diff
@@ -873,6 +875,11 @@ def format_array(values, formatter, float_format=None, na_rep='NaN',
         When formatting an Index subclass
         (e.g. IntervalIndex._format_native_types), we don't want the
         leading space since it should be left-aligned.
+    max_colwidth: False, int or None, optional, default None
```
Comment (jreback) [inline pandas/io/formats/format.py]: can you add a versionadded tag

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, can you add a whatsnew note

**Comment by jreback** [comment]
@simonjayhawkins can you rebase and update when possible

**Comment by jreback** [comment]
@simonjayhawkins whats the status of this?

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins whats the status of this?

still need to do a precursor. checks were passing and i don't think they should have been. so holding as draft unless you'd rather it be closed for now.

> @simonjayhawkins can you rebase and update when possible

comments addressed.

> looks good, can you add a whatsnew note

this should be a pure refactor. no whatsnew needed?

File: pandas/io/formats/format.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/io/formats/format.py]: not a public function. have added anyway.

File: pandas/io/formats/format.py [hunk 3]
```diff
@@ -420,6 +428,24 @@ def _get_formatter(self, i):
                 i = self.columns[i]
             return self.formatters.get(i, None)
 
+    def _format_col(self, i: int):
+        """
+        Calls `format_array` for column `i` of truncated DataFrame with
```
Comment (jreback) [inline pandas/io/formats/format.py]: can you say positional column i

File: pandas/io/formats/format.py [hunk 4]
```diff
@@ -988,14 +1003,21 @@ def format_array(
         When formatting an Index subclass
         (e.g. IntervalIndex._format_native_types), we don't want the
         leading space since it should be left-aligned.
+    max_colwidth: False, int or None, optional, default None
+        Whether the array should be formatted with strings truncated.
+        * False: do not truncate strings
+        * int: the maximum width of strings
+        * None: use display.max_colwidth setting
+
+        .. versionadded:: 1.0
```
Comment (jreback) [inline pandas/io/formats/format.py]: can just remove the versionadded (or make it 0.25.0)

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine. small comments.

**Comment by simonjayhawkins** [comment]
can close for now.

**Pull request closed without merging**